articles/create-package.md

```diff
@@ -5,18 +5,92 @@ title: "Creating a Package"
+    [WorkflowElementCategory(ElementCategory.Source)]
+    public class Source1
+    {
+        public IObservable<double> Generate()
+        {
+            return Observable
+                .Timer(
+                    dueTime: TimeSpan.Zero,
+                    period: TimeSpan.FromSeconds(0.5))
+                .Select(counter => Math.Sin(counter));
+        }
+    }
+    ```
+
+5. We can test our operator by starting the project with **F5**. This will automatically launch the Bonsai editor with our package preloaded in the **Toolbox**. Add the new `Source1` node and run the workflow.
+
+    ![Running the sine source](~/images/extensions-running.png)
+
+6. If we need to parameterize our operator, we can add new public properties to the class. Custom editors can be used to provide interactive widgets for editing the property values. Several common widgets are provided in the <xref href="Bonsai.DesignTypes"/> class. For example, to make the period customizable with a slider over a specified range, we can modify our source operator:
+
+    ```c#
+    [Description("")]
+    [Combinator(MethodName = nameof(Generate))]
+    [WorkflowElementCategory(ElementCategory.Source)]
+    public class Source1
+    {
+        [Range(0.1, 2)]
+        [Editor(DesignTypes.SliderEditor, DesignTypes.UITypeEditor)]
+        public double PeriodSeconds { get; set; } = 0.5;
+
+        public IObservable<double> Generate()
+        {
+            return Observable
+                .Timer(
+                    dueTime: TimeSpan.Zero,
+                    period: TimeSpan.FromSeconds(PeriodSeconds))
+                .Select(counter => Math.Sin(counter));
+        }
+    }
+    ```
+
+7. It is also possible to debug our code while it is running by setting breakpoints, either by clicking on the left of the line we want to debug or by hitting **F9** over the target code. After this, you can run the code step by step and inspect the runtime value of variables.
+
+    ![Debugging the sine source](~/images/extensions-debugging.png)
+
+## Publishing a package project
+
+1. Double-click the name of the project in the Visual Studio Solution Explorer to open up the package metadata.
+
+    ![Inspecting the package metadata](~/images/extensions-packagemetadata.png)
+
+2. Fill in or edit all the relevant metadata fields. These are critical to correctly communicate the provenance of your project to other users. Please pay special attention to `Title`, `Description`, `Authors`, `Copyright`, `PackageProjectUrl`, `PackageLicenseExpression`, `PackageIcon` and `PackageTags` to make sure that they correctly describe your project. Make sure that `Version` is correctly assigned in every new release to avoid problems during package updates.
+
+    > [!Tip]
+    > Use version suffixes for sharing prerelease versions for testing, e.g. `0.1.0-alpha`. If a package version has a prerelease suffix, it will only be listed by the package manager if the checkbox "Include prerelease" is checked.
```

Comment:
I would probably add something about how if you have the package currently installed in your Packages folder and try to compile a package with the same name and version, it will default to the one previously installed and you wont be able to use the just compiled package. Which reminds me, we should revisit this behavior.